contracts/RToken.sol

```diff
@@ -1012,4 +1015,117 @@ contract RToken is
             account.rInterest = rGross - account.lDebt;
         }
     }
+
+
+    /// @dev IRToken.mintFor implementation
+    function mintFor(uint256 mintAmount, address holder, address spender, uint256 nonce, uint256 expiry,
+                    bool allowed, uint8 v, bytes32 r, bytes32 s)
+        external
+        nonReentrant
+        returns (bool)
+    {
+        require(spender == address(this), "Spender must be this contract.");
+        Dai daiToken = Dai(ias.underlying());
+
+        // Call DAI.permit - this will allow this contract to pull DAI from user & will fail if not valid signature
+        // permit required each time???
+        daiToken.permit(holder, spender, nonce, expiry, allowed, v, r, s);
```

Comment:
Let's have an interface `IPermisibleToken` for example, instead of using the full DAI contract